--- a/tools/lldb/scripts/Python/finish-swig-Python-LLDB.sh
+++ b/tools/lldb/scripts/Python/finish-swig-Python-LLDB.sh
@@ -118,6 +118,8 @@
     echo "Python files will be put in ${framework_python_dir}"
 fi
 
+framework_python_dir=$DESTDIR/$framework_python_dir
+
 python_dirs="${framework_python_dir}"
 
 for python_dir in $python_dirs
@@ -157,7 +159,7 @@
     then
         ln -s "../../../LLDB" _lldb.so
     else
-        ln -s "../../../liblldb${SOEXT}" _lldb.so
+        ln -s "../../../llvm/liblldb${SOEXT}" _lldb.so
     fi
 else
     if [ $Debug -eq 1 ]
